### PR TITLE
Prepare to move --delete-tostring-package-uri option to Dart SDK

### PR DESCRIPTION
**Pull request by alexmarkov**
This change prepares Flutter front-end server to the migration of `--delete-tostring-package-uri` option to Dart SDK. This is needed to call toString transformation before tree shaking.

https://github.com/dart-lang/sdk/issues/46022

### CODE REVIEW

File: flutter_frontend_server/lib/server.dart [hunk 1]
```diff
@@ -111,13 +111,18 @@ Future<int> starter(
   frontend.ProgramTransformer transformer,
 }) async {
   ArgResults options;
-  frontend.argParser.addMultiOption(
-    'delete-tostring-package-uri',
-    help: 'Replaces implementations of `toString` with `super.toString()` for '
+  final bool handleDeleteToString =
+      !frontend.argParser.options.containsKey('delete-tostring-package-uri');
+  if (handleDeleteToString) {
```
Comment (dnfield) [inline flutter_frontend_server/lib/server.dart:116]: Won't this always evaluate to false?

Comment (alexmarkov) [reply flutter_frontend_server/lib/server.dart:116]: `frontend.argParser.options.containsKey('delete-tostring-package-uri')` will be true when this option is added to the front-end server in Dart SDK. At that point we should stop handling it here in the Flutter engine. This allows us to have a seamless migration.

Comment (dnfield) [reply flutter_frontend_server/lib/server.dart:116]: Ahhh I understand now, we're looking at a different arg parser we don't own. Ok, LGTM

**Review by dnfield — APPROVED** [review]

File: flutter_frontend_server/lib/server.dart (hunk 1 above)
Comment (dnfield) [reply flutter_frontend_server/lib/server.dart:116]: Perhaps as a nit add a TODO about when we can remove this with al ink to an issue

**Pull request merged**